fineract-provider/src/main/java/org/apache/fineract/useradministration/service/AppUserWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -349,6 +349,10 @@ private void handleDataIntegrityIssues(final JsonCommand command, final Throwabl
                     username);
         }
 
+        if (realCause.getMessage().contains("'unique_self_client'")) {
```

Comment:
Depending on the message of an exception containing a specific hard coded String is not great... the way one would usually do something like this in Java is to check for ‘instanceof‘ a particular (sub)class of an Exception. Could you do that here?

Reply:
It is the same method used here to check for duplicate username, please check line number 344, a few lines above this.
hard coded string is the unique key for that column.

Reply:
I do agree with @vorburger . 
Correct me if I am wrong.
The function is called from 4 places in this file and all of them are within a catch block.
Can you refactor this file a bit, to avoid doing redundant `if-else` as the catch block identifies which type of Exception is there( in a way it is also having that `if-else` logic) ?